File: JS/DecodeTheSlantedCiphertext.test.js
/**
 * @version JAVASCRIPT ECMAScript 6 
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { decodeCiphertext } from './DecodeTheSlantedCiphertext.js'

TEST('Starting DecodeTheSlantedCiphertext test...', (t) => {
    //Test 1
    let want = "cipher"
    let got = decodeCiphertext( "ch   ie   pr", 3 )
    t.assert( want === got, "Expect: "+want)
    //Test 2
    want = "i love leetcode"
    got = decodeCiphertext( "iveo    eed   l te   olc", 4 )
    t.assert( want === got, "Expect: "+want)
    //Test 3
    want = "coding"
    got = decodeCiphertext( "coding", 1 )
    t.assert( want === got, "Expect: "+want)

    t.end()
})

File: GO/DecodeTheSlantedCiphertext_test.go
/**
 * @version GO 1.22.3
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestDecodeTheSlantedCiphertext
 */
package GO

import (
	"reflect"
	"testing"
)

func TestDecodeTheSlantedCiphertext(t *testing.T) {
	tables := []struct {
		want string
		nums string
		k    int
	}{
		{"cipher", "ch   ie   pr", 3},
		{"i love leetcode", "iveo    eed   l te   olc", 4},
		{"coding", "coding", 1},
	}

	for _, table := range tables {
		got := decodeCiphertext(table.nums, table.k)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
